TPP.Core/Moderation/ModerationService.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        if (targetUser.Roles.Overlaps(new[] { Role.Operator, Role.Moderator }))
+            return BanResult.UserIsModOrOp;
+
+        Instant now = _clock.GetCurrentInstant();
+
+        if (isBan)
+            await _executor.Ban(targetUser, reason);
+        else
+            await _executor.Unban(targetUser, reason);
+
+        await _banLogRepo.LogBan(
+            targetUser.Id, isBan ? "manual_ban" : "manual_unban", reason,
+            issuerUser.Id, now);
+        await _timeoutLogRepo.LogTimeout( // bans/unbans automatically lift timeouts
+            targetUser.Id, isBan ? "untimeout_from_manual_ban" : "untimeout_from_manual_unban", reason,
+            issuerUser.Id, now, null);
+        await _userRepo.SetBanned(targetUser, isBan);
+
+        ModerationActionPerformed?.Invoke(this, new ModerationActionPerformedEventArgs(
+            issuerUser, targetUser, isBan ? ModerationActionType.Ban : ModerationActionType.Unban));
+
+        // TODO is this obsolete?
+        await _messageSender.SendWhisper(targetUser,
+            $"You were {(isBan ? "banned" : "unbanned")} " +
+            $"by {issuerUser.Name} at {now}. Reason: {reason}");
+
+        return BanResult.Ok;
+    }
+
+    public Task<TimeoutResult> Timeout(User issuerUser, User targetUser, string reason, Duration duration) =>
+        TimeoutOrUntimeout(issuerUser, targetUser, reason, duration);
+    public Task<TimeoutResult> Untimeout(User issuerUser, User targetUser, string reason) =>
+        TimeoutOrUntimeout(issuerUser, targetUser, reason, null);
+
+    private async Task<TimeoutResult> TimeoutOrUntimeout(
+        User issuerUser, User targetUser, string reason, Duration? duration)
+    {
+        bool isIssuing = duration != null;
+        if (duration.HasValue && duration > Duration.FromDays(14))
+            return TimeoutResult.MustBe2WeeksOrLess;
+        if (targetUser.Banned)
+            return TimeoutResult.UserIsBanned;
+
+        Instant now = _clock.GetCurrentInstant();
+
+        if (isIssuing)
+            await _executor.Timeout(targetUser, reason, duration!.Value);
+        else
+            await _executor.Unban(targetUser, reason);
+        await _timeoutLogRepo.LogTimeout(
+            targetUser.Id, isIssuing ? "manual_timeout" : "manual_untimeout", reason,
+            issuerUser.Id, now, duration);
+        await _userRepo.SetTimedOut(targetUser, duration.HasValue ? now + duration.Value : null);
+
+        ModerationActionPerformed?.Invoke(this, new ModerationActionPerformedEventArgs(
+            issuerUser, targetUser, isIssuing ? ModerationActionType.Timeout : ModerationActionType.Untimeout));
+
+        // TODO is this obsolete?
+        await _messageSender.SendWhisper(targetUser,
+            $"You were {(isIssuing ? "timed out" : "untimed out")} " +
+            $"by {issuerUser.Name} at {now}. Reason: {reason}");
+
+        return TimeoutResult.Ok;
```

Comment:
Users already get timeout and ban reasons displayed in chat. Is this still needed? Only advantage I see is that they also see any potential _unban_ or _untimeout_ reasons.

Reply:
I assumed it's unnecessary and removed it